### PR TITLE
[salesforce] - Verify the `instanceUrl` is a valid Salesforce domain

### PR DESCRIPTION
**Pull request by nick-Ag**
This PR verifies the Salesforce `instanceUrl` is a valid `salesforce.com` domain before sending a request to the url.
Addresses: https://segment.atlassian.net/browse/STRATCONN-1270

## Testing

Tested in staging and locally.
Overrode `instanceUrl` to a non-salesforce domain:
Screenshot 2023-01-31 at 3 08 58 PM

Requests made fail as expected:
Screenshot 2023-01-31 at 3 11 08 PM

Does not fail when the `instanceUrl` is a Salesforce domain:
Screenshot 2023-01-31 at 3 12 36 PM

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [x] [Segmenters] Tested in the staging environment

### CODE REVIEW

File: packages/destination-actions/src/destinations/salesforce/sf-utils.ts [hunk 1]
```diff
@@ -172,3 +172,26 @@ const snakeCaseToPascalCase = (key: string): string => {
   const token = camelCase(key)
   return token.charAt(0).toUpperCase() + token.slice(1)
 }
+
+export const validateInstanceURL = (instanceUrl: string): string => {
+  if (instanceUrl === undefined || instanceUrl === '') {
```
Comment (silesky) [inline packages/destination-actions/src/destinations/salesforce/sf-utils.ts:177]: if (!instanceUrl) covers '', undefined, null -- or, are you saying that null is a valid value?

File: packages/destination-actions/src/destinations/salesforce/sf-utils.ts [hunk 2]
```diff
@@ -172,3 +172,26 @@ const snakeCaseToPascalCase = (key: string): string => {
...
+    throw new IntegrationError(
+      'Empty Salesforce instance URL. Please login through OAuth.',
+      'INVALID_INSTANCE_URL',
+      400
+    )
+  }
+
+  const salesforceRegex = /^(https?):\/\/.*\.salesforce\.com/
```
Comment (marinhero) [inline packages/destination-actions/src/destinations/salesforce/sf-utils.ts:185]: 👍

**Review by marinhero — APPROVED** [review]

**Review by marinhero — DISMISSED** [review]
oops! Fix the failing test and we can 🚢

File: packages/destination-actions/src/destinations/salesforce/sf-utils.ts (hunk 2 above)
Comment (mericsson) [reply packages/destination-actions/src/destinations/salesforce/sf-utils.ts:185]: Have we queried the DB to see that this validates correctly for all expected customer URLs?

Comment (mericsson) [reply packages/destination-actions/src/destinations/salesforce/sf-utils.ts:185]: Also, I think we should enforce `https`

**Review by marinhero — APPROVED** [review]
Thanks for addressing the comments and checking the DB. :shipit:

**Pull request merged**